packages/analytics/analytics-geo-map/src/components/AnalyticsGeoMap.vue

```diff
@@ -0,0 +1,244 @@
+<template>
+  <div
+    id="mapContainer"
+    ref="mapContainer"
+    class="kong-ui-public-analytics-geo-map"
+  />
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, watch, computed } from 'vue'
+import { Map, Popup } from 'maplibre-gl'
+import type { PropType } from 'vue'
+import type { ColorSpecification, DataDrivenPropertyValueSpecification, ExpressionSpecification, LngLatLike, MapOptions } from 'maplibre-gl'
+import type { LongLat, MapFeatureCollection } from '../types'
+import type { FeatureCollection, Feature, MultiPolygon } from 'geojson'
+import 'maplibre-gl/dist/maplibre-gl.css'
+
+const props = defineProps({
+  countryMetrics: {
+    type: Object as PropType<Record<string, number>>,
+    required: true,
+  },
+  geoJsonData: {
+    type: Object as PropType<MapFeatureCollection>,
+    required: true,
+  },
+  center: {
+    type: Object as PropType<LongLat>,
+    required: false,
+    default: null,
+  },
+  fitToCountry: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  initialZoom: {
+    type: Number,
+    required: false,
+    default: null,
+    validator: (value: number) => value >= 0 && value <= 24,
+  },
+  metricUnit: {
```

Comment:
question: should this prop have a validator or is literally any string acceptable?

Reply:
any string is not acceptable, it should be a number between 0 and 24 or null if the prop is not provided.